snapcraft/internal/lifecycle.py

```diff
@@ -252,6 +253,14 @@ def _create_meta(self, step, part_names):
             meta.create_snap_packaging(self.config.data,
                                        self.project_options)
 
+    def _record_snapcraft(self, step, part_names):
```

Comment:
I moved it to meta.py. But lets discuss it tomorrow in the standup, because I think we can do better.
I would like to split the logic that puts things in prime/meta from the one that puts things in prime/snap. 

Hooks, .desktop and icon are taken from $PROJECT_ROOT/snap and put into prime/meta. For now, I would leave those functions in meta.py. But we are calling them assets, maybe assets.py could work too.